packages/circuits/crates/libs/greco/src/lib.nr

```diff
@@ -233,27 +233,27 @@ impl<let N: u32, let L: u32> Greco<N, L> {
     ///
     /// # Returns
     /// An array containing all polynomials coefficients in flattened form
-    fn payload<let SIZE: u32>(self) -> [Field; SIZE] {
-        let mut inputs = [0; SIZE];
-        let mut offset = 0;
+    fn payload(self) -> Vec<Field> {
+        let mut inputs = Vec::new();
 
         // Flatten public key and ciphertext polynomials first (public inputs)
-        let (inputs, offset) = flatten(inputs, self.pk0is, offset);
-        let (inputs, offset) = flatten(inputs, self.pk1is, offset);
-        let (inputs, offset) = flatten(inputs, self.ct0is, offset);
-        let (inputs, offset) = flatten(inputs, self.ct1is, offset);
+
+        inputs = flatten::<_, _, BIT_PK>(inputs, self.pk0is);
+        inputs = flatten::<_, _, BIT_PK>(inputs, self.pk1is);
+        inputs = flatten::<_, _, BIT_PK>(inputs, self.ct0is);
+        inputs = flatten::<_, _, BIT_PK>(inputs, self.ct1is);
```

Comment:
why do we have the same `BIT_PK` for `pk` and `ct`? if it's the same size, can we rename it to something meaningful for both?

Reply:
Instead, I added `BIT_CT` to avoid confusion there. I believe this will be a better solution to increase readability since it won't affect anything in the circuit. Wdyt?